### PR TITLE
added DAO and Service layers for Order, ShoppingCart and User models

### PR DESCRIPTION
**Pull request by KseniiaMakarova**

### CODE REVIEW

File: src/main/java/com/internet/shop/service/ShoppingCartService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+
+public interface ShoppingCartService {
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart); //remove all products from the shoppingCart
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/ShoppingCartService.java:12]: коментар можна видалити

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Product> products, User user) {
+        List<Product> productsToOrder = new ArrayList<>(products);
+        shoppingCartService.clear(shoppingCartService.getByUserId(user.getId()));
+        Order newOrder = new Order(productsToOrder, user);
+        orderDao.create(newOrder);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java:27]:
```suggestion
        return orderDao.create(newOrder);
```

File: src/main/java/com/internet/shop/service/ShoppingCartService.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/service/ShoppingCartService.java:12]: sure, thank you)

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: done!

**Pull request merged**